contest/contest_info.py

```diff
@@ -0,0 +1,156 @@
+            contestant_submission_list.append(submission)
+        
+        submission_list.append({'contestant_submission_list':contestant_submission_list})
+        contestant_submission_list = []
+    return submission_list
+
+def get_penalty_scoreboard(contest):
+    
+    contestant_list = get_contestant_list(contest)
+    submission_list = get_contest_submission_list(contest)
+    
+    #store all contestants' data
+    scoreboard = []
+    #store single contestant score
+    contestant_scoreboard = []
+    #store single problem status (pass or not)
+    status = {}
+    #store single problem submit times
+    times = {}
+    #store single problem penalty
+    penalty = {}
+
+    wrong_submission_panelty = 20
+
+    #get single contestant
+    for contestant_submission_list in submission_list:
+        times = {}
+        status = {}
+        #count penalty
+        total_penalty = 0
+        #count solved
+        solved = 0
+        contestant = []
+        #get single contestant's submissions
+        for submissions in contestant_submission_list['contestant_submission_list']:
+            #get single contestant's single submission
+            for submission in submissions:
+                #count single problem submission times
+                times[submission.problem.id] = times.get(submission.problem.id,0) + 1
+                if submission.status == Submission.ACCEPTED:
+                    #status[problem name] = 1 if problem is solved
+                    status[submission.problem.id] = 1
+                    #get single problem penalty
+                    penalty[submission.problem.id] = penalty.get(submission.problem.id,0) + (submission.submit_time - contest.start_time).total_seconds()/60
+                else:
+                    # 20 is penalty if wrong submission
+                    penalty[submission.problem.id] = penalty.get(submission.problem.id,0) + wrong_submission_panelty 
+                contestant = submission.user
+            #calculate penalty and solved
+            solved = 0
+            for problem in contest.problem.all():
+                if status.get(problem.id,0) == 1:
+                    solved += 1
+                    total_penalty += penalty[problem.id]
+
+
+            contestant_scoreboard = {'status':status,'times':times,'penalty':total_penalty,'solved':solved}
+            scoreboard.append({'contestant':contestant,'contestant_scoreboard':contestant_scoreboard})
+        contestant_scoreboard = []
+    return scoreboard
+
+def get_testcase_scoreboard(contest):
+    
+    submission_list = get_contest_submission_list(contest)
+    
+    testcase_scoreboard = {}
+    # store single contestant testcase-based score
+    contestant_testcase_scoreboard = {}
+    # store single problem passed testcase number
+    testcase_status = {}
+    # store single problem total testcase number
+    testcase_number = {}
+    if submission_list.__len__() != 0:
```

Comment:
I think it's unnecessary to check this. If the list is empty, for loop will terminate immediately.